### PR TITLE
Further donation modal tweaks

### PR DESCRIPTION
**Pull request by dem4ron**

### CODE REVIEW

File: app/helpers/react_components/student/mentoring_session.rb [hunk 1]
```diff
@@ -23,8 +23,6 @@ def to_h
           out_of_date: solution.out_of_date?,
           videos:,
           donation: {
-            user_signed_in: true,
-            captcha_required: false,
             recaptcha_site_key: Exercism.secrets.recaptcha_site_key,
```
Comment (iHiD) [inline app/helpers/react_components/student/mentoring_session.rb]: I feel like we shouldn't need this if we don't need the two above?

File: app/javascript/components/modals/student/FinishMentorDiscussionModal.tsx [hunk 2]
```diff
@@ -79,16 +83,20 @@ const Inner = ({
       if (donation.showDonationModal) {
         return (
           <Step.DonationStep
-            exerciseLink={links.exercise}
             donation={donation}
             links={links}
+            onSuccessfulDonation={(_, amount) => {
+              send('SUCCESSFUL_DONATION')
+              setDonatedAmount(amount)
```
Comment (dem4ron) [inline app/javascript/components/modals/student/FinishMentorDiscussionModal.tsx]:
```suggestion
              setDonatedAmount(amount)
              send('SUCCESSFUL_DONATION')
```

File: app/javascript/components/modals/student/finish-mentor-discussion-modal/CelebrationStep.tsx [hunk 3]
```diff
@@ -1,30 +1,33 @@
...
+    return () => setContainerModalMaxWidth('100%')
+  }, [setContainerModalMaxWidth])
+
   return (
     <section className="celebrate-step neon-cat">
-      <img
-        src="https://media.giphy.com/media/sIIhZliB2McAo/source.gif"
-        className="gif"
-      />
+      <img src="https://i.gifer.com/17xo.gif" className="gif" height={80} />
```
Comment (dem4ron) [inline app/javascript/components/modals/student/finish-mentor-discussion-modal/CelebrationStep.tsx]:
```suggestion
      <img src="https://i.gifer.com/17xo.gif" className="gif"/>
```

File: app/javascript/components/modals/student/FinishMentorDiscussionModal.tsx [hunk 4]
```diff
@@ -133,22 +149,28 @@ export const FinishMentorDiscussionModal = ({
...
   return (
     <Modal
-      style={{ content: { maxWidth: '100%' } }}
+      style={{ content: { maxWidth } }}
       cover
       aria={{ modal: true, describedby: 'a11y-finish-mentor-discussion' }}
       className="m-finish-student-mentor-discussion"
       ReactModalClassName="bg-unnamed15"
-      shouldCloseOnOverlayClick={false}
+      shouldCloseOnOverlayClick
```
Comment (dem4ron) [inline app/javascript/components/modals/student/FinishMentorDiscussionModal.tsx]:
```suggestion
      shouldCloseOnOverlayClick={false}
```

**Pull request merged**